plugins/notifications-node/src/extensions.ts

```diff
@@ -100,13 +100,37 @@ export interface NotificationProcessor {
   getNotificationFilters?(): NotificationProcessorFilters;
 }
 
+/**
+ * NotificationRecipientResolver interface is used to resolve the individual
+ * users to receive the notification based on the entity reference(s) and
+ * the excluded entity reference(s).
+ *
+ * Using this extension point replaces the default recipient resolving
+ * functionality of the notifications backend.
+ *
+ * The `resolveNotificationRecipients` should return object with a list of user
+ * entity references that should receive the notification. In case the function
+ * returns other than user entity references, those are ignored.
```

Comment:
Reading this comment, I feel some of it should be on this type, some of it should be on `resolveNotificationRecipients`, and some of it should be on `setNotificationRecipientResolver`.